### PR TITLE
feat(console): component definition write

### LINKED ISSUE #610: Console: Component Definition Write
### Is your feature request related to a problem? Please describe.

When a user has a component definition loaded in the Console, they should be able to update certain fields, such as the description and remarks data.

### Describe the solution you'd like

- **Given** console is opened with oscal model
- **When** a user edits a field
- **Then** edited data is written back to oscal model

### Describe alternatives you've considered

none

### Additional context

This will require some additional features, such as a "save" option and warning on close, such that the user is given an option to write any unsaved data. 

### Expected Deliverable

Console functionality updates of "save" and "warn". Documentation for feature.

### PR DESCRIPTION
**Pull request by meganwolf0**
## Description

updates to allow for write of the `remarks` and `description` fields of the component definition. Other "quality of life" updates with respect to simplifying views also incorporated.

## Related Issue

Fixes #610 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Schema Updates](https://github.com/defenseunicorns/lula/blob/main/docs/community-and-contribution/schema-updates.md) applied
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/lula/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: docs/console/component-definition.md [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-The Component Definition view currently allows for a read-only experience of the OSCAL Component Definition model. The view is tailored to the usage of the Component Definition in the context of Lula, and is not intended to be a comprehensive view of the model.
+The Component Definition view currently allows for a read and limited write experience of the OSCAL Component Definition model. The view is tailored to the usage of the Component Definition in the context of Lula, and is not intended to be a comprehensive view of the model.
 
 ## Usage
 
 To view an OSCAL Component Definition model in the Console:
 ```shell
 lula console -f /path/to/oscal-component.yaml
 ```
+The `oscal-component.yaml` will need to be a valid OSCAL model - to use with the Component Definition view, it must contain the `component-definition`.
```
Comment (brandtkeller) [inline docs/console/component-definition.md]:
```suggestion
The `oscal-component.yaml` will need to be a valid OSCAL model - to use with the Component Definition view, it must contain the `component-definition` top level key.
```

Or something similar

File: src/cmd/console/console.go [hunk 2]
```diff
@@ -46,15 +46,20 @@ var consoleCmd = &cobra.Command{
 
 		// Add debugging
 		// TODO: need to integrate with the log file handled by messages
+		var dumpFile *os.File
 		if message.GetLogLevel() == message.DebugLevel {
-			f, err := tea.LogToFile("debug.log", "debug")
+			// f, err := tea.LogToFile("debug.log", "debug")
```
Comment (brandtkeller) [inline src/cmd/console/console.go]: Intending to keep this code?

**Review by brandtkeller — COMMENTED** [review]
These are some massive PR's but totally understand there may not be a better way to slice them. 

Making my way through currently - Don't want to needlessly block these but still processing how best to review (and have the team review)

**Review by brandtkeller — COMMENTED** [review]
stoked about the functionality and seriously think there is huge long-term value for iteration. 

I ran into an inconsistency once where I made an edit -> saved -> attempted to quit and was told there was pending changes to save -> couldn't reproduce it though but now understand that save check even better. 

There are minor outstanding comments - If we resolve then I will do a final review.

File: docs/console/component-definition.md (hunk 1 above)
Comment (meganwolf0) [reply docs/console/component-definition.md]: updated

File: src/cmd/console/console.go (hunk 2 above)
Comment (meganwolf0) [reply src/cmd/console/console.go]: removed

File: src/internal/tui/model.go [hunk 3]
```diff
@@ -90,9 +155,62 @@ func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 				m.activeTab = m.activeTab - 1
 			}
 
+		case common.ContainsKey(k, m.keys.Confirm.Keys()):
+			if m.closeModel.Open {
+				return m, tea.Quit
+			}
+
+		case common.ContainsKey(k, m.keys.Save.Keys()):
+			m.saveModel.Open = true
```
Comment (brandtkeller) [inline src/internal/tui/model.go:174]: I believe you'll want some additional handling in this to account for:

- Edit a compdef remarks field for any given control
- Attempt to quit with `ctrl+c`
- The `closeModel` will notify you that your `changes not written`
- Attempt to save with `ctrl+s` (this does nothing currently)

**Review by brandtkeller — CHANGES_REQUESTED** [review]

File: src/internal/tui/model.go (hunk 3 above)
Comment (meganwolf0) [reply src/internal/tui/model.go:174]: @brandtkeller sorry I'm not following all the keystroke use cases here, lmk if you have time to sync up and walk me through the operations you are trying to satisfy

Comment (brandtkeller) [reply src/internal/tui/model.go:174]: workflow addressed

**Review by brandtkeller — APPROVED** [review]
Outstanding work - appreciate the significance in enabling what will become essential to adoption.

**Pull request merged**